src/extra/AutoUI/utils.ts

```diff
@@ -71,3 +77,41 @@ export const autoUIGetDisabledReason = <T extends AutoUIBaseResource<T>>(
 		});
 	}
 };
+
+export const formatSorters: Dictionary<TableSortFunction<any>> = {};
+
+export const getSortingFunction = <T extends any>(
+	schemaKey: keyof T,
+	schemaValue: JSONSchema,
+): TableSortFunction<T> => {
+	if (formatSorters[schemaValue.format ?? '']) {
+		return formatSorters[schemaValue.format ?? ''];
+	}
+
+	const types = castArray(schemaValue.type);
+	if (types.includes(JsonTypes.string)) {
+		return (a: T, b: T) => {
+			const aa = (a[schemaKey] ?? '') as string;
+			const bb = (b[schemaKey] ?? '') as string;
+			if (typeof aa === 'string' && typeof bb === 'string') {
+				return aa.toLowerCase().localeCompare(bb.toLowerCase());
+			}
+			return diff(aa, bb);
+		};
+	}
+	return (a: T, b: T) =>
+		// @ts-expect-error
+		diff(a[schemaKey], b[schemaKey]);
```

Comment:
same as above

Reply:
> Another instance of pre-existing code @JSReds. I've changed the code as per your suggestion but I'm not 100% sure I did what you had in mind, could you please check it out and confirm?

Same here, could you please confirm?

Reply:
I was referring to the @ts-expect-error, this is clearly something forced here.  let's leave it like this I'll refactor this part as soon as I can